### PR TITLE
[DOMAIN] Member DOMAIN BASIC

### PR DESCRIPTION
**Pull request by DongGeon0908**
Basic Member Domain 입니다~

### CODE REVIEW

File: src/main/java/com/excmul/member/domain/vo/BirthVo.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @Column(name = "MEMBER_BIRTH", nullable = false)
+    private String birth;
+
+    public BirthVo(String birth) {
+        validate(birth);
+        this.birth = birth;
+    }
+
+    public void validate(String birth) {
+        if (!Pattern.matches(BIRTH_VALIDATOR, birth)) {
```
Comment (darkant99) [inline src/main/java/com/excmul/member/domain/vo/BirthVo.java:35]: StringUtils.hasText

File: src/main/java/com/excmul/member/domain/vo/PasswordVo.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    private static final String PASSWORD_VALIDATOR = "^(?=.*[A-Za-z])(?=.*\\d)(?=.*[$@$!%*#?&])[가-힣A-Za-z\\d$@$!%*#?&]{8,30}$";
+
+    @Column(name = "member_password", nullable = false)
+    private String password;
+
+    public PasswordVo(String password) {
+        validate(password);
+        this.password = passwordEncode(password);
+    }
+
```
Comment (darkant99) [inline src/main/java/com/excmul/member/domain/vo/PasswordVo.java:41]: private PasswordVo(String password) {
        validate(password);
        this.password = password;
    }

public static PasswordVo of(String password) {
return new PasswordVo(passwordEncode(password));
}

File: src/main/java/com/excmul/member/dto/DefaultMemberSignRequest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.excmul.member.domain.*;
+import com.excmul.member.domain.vo.*;
+import lombok.*;
+
+@Setter
+@Getter
+@NoArgsConstructor
+public class DefaultMemberSignRequest {
+
```
Comment (darkant99) [inline src/main/java/com/excmul/member/dto/DefaultMemberSignRequest.java:11]: BasicMembeSignRequest ??

File: src/main/java/com/excmul/member/exception/MemberExceptionMessage.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    EMAIL("이메일 형식이 맞지 않습니다."),
+    NICKNAME("닉네임 형식이 맞지 않습니다."),
+    PHONE_NUMBER("전화번호 형식이 맞지 않습니다."),
+    PASSWORD("비밀번호는 최소 8자 ~ 최대 30자, 최소 영문 소문자, 대문자, 특수문자를 각각 하나 이상 포함해야 합니다."),
+    BIRTH("생년월일은 yyyymmdd 형식에 맞추어 작성해주세요."),
+    DUPLICATION_EMAIL("중복된 이메일 입니다."),
+    DUPLICATION_NICKNAME("중복된 닉네임 입니다."),
+    DUPLICATION_PHONE_NUMBER("중복된 전화번호 입니다."),
+    FAIL_SIGN("회원가입을 실패했습니다.");
+
```
Comment (darkant99) [inline src/main/java/com/excmul/member/exception/MemberExceptionMessage.java:14]: Exception한테 각각 역할 위임

File: src/main/java/com/excmul/member/domain/vo/BirthVo.java (hunk 1 above)
Comment (DongGeon0908) [reply src/main/java/com/excmul/member/domain/vo/BirthVo.java:35]: 수정 완료했습니다~

File: src/main/java/com/excmul/member/dto/DefaultMemberSignRequest.java (hunk 3 above)
Comment (DongGeon0908) [reply src/main/java/com/excmul/member/dto/DefaultMemberSignRequest.java:11]: 명칭 변경했습니다!

**Pull request closed without merging**